### ISSUE TITLE
Add array safe cast argument to function declarations.

### ISSUE THREAD

**Issue by kse300489** [zephir-lang/zephir#811]
Right now not matter what sintax we use:

``` php
public function method000(array arr){

}
```

or

``` php
public function method000(array! arr){

}
```

If we pass not array to `method000`, we get error.

``` text
PHP Catchable fatal error:  Argument 1 passed to %%%::method000() must be of the type array, %%% given in
```

Maybe add auto array safe cast in first example:

``` php
public function method000(array arr){
 var_dump(arr);
}
```

And if we pass not array to this method, zephir automatically wrap passed variable to array.
Example:

``` php
$obj->method000(1);
```

Output:

``` Text
array(1) {
  [0] =>
  int(1)
}
```
##

**Comment by ovr** [comment]
So I think It cant be implemented via Array type declaration is worked in PHP Core
I mean
ZEND_ARG_INFO_ARRAY
And this error

```
PHP Catchable fatal error:  Argument 1 passed to %%%::method000() must be of the type array, %%% given in
```

Is caused from Core before php function will be called

**Comment by serg-deep** [comment]
I know about ZEND_ARG_INFO_ARRAY.

It can be used if we use strict type checking. Like this:
```php
public function method000(array! argument){
}
```
But maybe it can be implemented on zephir type checking layer. Like this:
```php
public function methodBool(bool argument){
  var_dump(argument);
}
```
Zephir automatically cast all not bool argumets to `bool`. So if we run this code:
```php
$this->methodBool('');
$this->methodBoo(0);
$this->methodBool(NULL);
$this->methodBool(FALSE);
```
We get ouput:
```text
bool(false)
bool(false)
bool(false)
bool(false)
```

Also and this case:
```php
public function methodArray(array argument){
  var_dump(argument);
}
```
Zephir will automatically cast all not `array` arguments to `array`. Example:
```php
$this->methodArray('');
$this->methodArray(0);
$this->methodArray(NULL);
$this->methodArray(FALSE);
$this->methodArray([]);
```
Output:
```text
array(1) {
  [0] =>
  string(0) ""
}
array(1) {
  [0] =>
  int(0)
}
array(1) {
  [0] =>
  NULL
}
array(1) {
  [0] =>
  bool(false)
}
array(0) {
}
```

Of сourse we can't use ZEND_ARG_INFO_ARRAY right now, but we can use simple ZEND_ARG_INFO(0, argument) and then check and cast input method arguments by generated C code.

**Comment by Jeckerson** [comment]
Implemented in `0.13.0` when using PHP8.
